pkg/liveness/liveness.go

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2018 Tigera, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package liveness
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+)
+
+var felixLivenessEp string
+
+func init() {
+	felixPort := os.Getenv("FELIX_HEALTHPORT")
```

Comment:
maybe this would be better as `os.LookupEnv` to distinguish from an explicit blank `FELIX_HEALTHPORT`?

Reply:
Good thing to think about but I can't see any reason to distinguish the two cases here; go idiom is to make the zero value useful. In this case no env var and env var equal to `""` can both mean "use the default"; WDYT?

Reply:
I was trying to imagine what a user would expect from `FELIX_HEALTHPORT="" node-calico [...]`

_(in practice, it would probably be the result of some other bad config.)_

Flagging empty when the `Atoi` fails just below might be friendlier than falling back on the default.

But I'm not against the blank field defaulting :smile:

Reply:
I think `os.Getenv` and `os.LookupEnv` would result in the same logic here since for our case we use the default for all cases except where the environment variable is set correctly, so I don't see a reason to use `os.LookupEnv` over `os.Getenv`